httpclient.go

```diff
@@ -63,20 +64,24 @@ func (c *HttpClientImplementation) Call(method string, url string, apiKey *strin
 	if apiKey != nil {
 		key := *apiKey
 		if key == "" {
+			errMessage := "The API Key (ServerKey/IrisApiKey) is invalid, as it is an empty string. Please double-check your API key. " +
+				"You can check from the Midtrans Dashboard. " +
+				"See https://docs.midtrans.com/en/midtrans-account/overview?id=retrieving-api-access-keys " +
+				"for the details or please contact us via https://midtrans.com/contact-us. "
 			err := &Error{
-				Message: "The API Key (ServerKey/IrisApiKey) is invalid, as it is an empty string. Please double-check your API key. " +
-					"You can check from the Midtrans Dashboard. " +
-					"See https://docs.midtrans.com/en/midtrans-account/overview?id=retrieving-api-access-keys " +
-					"for the details or contact support at support@midtrans.com if you have any questions.",
+				Message: errMessage,
+				RawError: errors.New(errMessage),
 			}
```

Comment:
Creating error this way seems redundant, `errMessage` duplicated in `Message & RawError` params, how about this, wdyt?

when Creating error:
```golang
err := &Error{
  Message: "my custom err msg string",
  // "RawError:" is optional, no need to supply RawError if there is no error happened 
  // from the underlying code/library/network/etc
}
```

in Error func, no need to check  & return the RawError
```golang

func (e *Error) Error() string {
  // if there is RawError supplied during error creation, 
  // return the custom error msg plus the RawError msg
  if e.RawError != nil {
    return fmt.Sprintf("%s: %s",e.Message,e.RawError.Error())
  }
  return e.Message
}
```

in Error Unwrap, return the RawError here (if any)
```golang
func (e *Error) Unwrap() error {
  // if there is RawError supplied during error creation, return RawError. Else, will return nil
  return e.RawError
}
```

Reply:
> Creating error this way seems redundant, errMessage duplicated in Message & RawError params, how about this, wdyt?

Yes Agreed, will be redundant. AFAIK but if the RawError is nil we can't use unwrap method. and will not be suitable to use error.Is because the error is will check raw error object


> in Error func, no need to check & return the RawError
> 
> func (e *Error) Error() string {
>   // if there is RawError supplied during error creation, 
>   // return the custom error msg plus the RawError msg
>   if e.RawError != nil {
>     return fmt.Sprintf("%s: %s",e.Message,e.RawError.Error())
>   }
>   return e.Message
> }

Seems make sense, will change to this

> in Error Unwrap, return the RawError here (if any)
> 
> func (e *Error) Unwrap() error {
>   // if there is RawError supplied during error creation, return RawError. Else, will return nil
>   return e.RawError
> }

I still don't get the point here. Do you mean to add this description to Unwrap function da? 
`// if there is RawError supplied during error creation, return RawError. Else, will return nil`

Reply:
@Xaxxis 
1. you don't need unwrap to test the error.Is, your current test is incorrect will leave separate comment to review that part.
2. 👌 
3. Actually no code change needed for that part, i'm just adding explanation because the RawError now can be `nil` because I mentioned `"RawError:" is optional` on point 1. Also yes, nice to add the comment/description.

Btw I [made a test/sample code here](https://replit.com/@raizerde/errorUnwrapAsIs#main.go), which you can try.